hydrocron_database/tests/test_hydrocron_database.py

```diff
@@ -0,0 +1,208 @@
+            TEST_TABLE_NAME,
+            DYNAMO_KEYS)
+
+        # read shapefile into geopandas dataframe
+        shp_file = gpd.read_file(TEST_SHAPEFILE_PATH)
+
+        item_attrs = {}
+        for _index, row in shp_file.iterrows():
+            # convert each reach into a dictionary of attributes
+            item_attrs = json.loads(row.to_json(
+                default_handler=str), parse_float=Decimal)
+
+            # write to the table
+            hydrocron_test_table.add_data(**item_attrs)
+
+    assert hydrocron_test_table.table.item_count == 687
+
+
+def test_query(dynamo_instance):
+    '''
+    Test a query for a reach id
+    '''
+
+    if dynamo_instance.table_exists(TEST_TABLE_NAME):
+        dynamo_instance.delete_table(TEST_TABLE_NAME)
+
+    hydrocron_test_table = dynamo_instance.create_table(
+        TEST_TABLE_NAME,
+        DYNAMO_KEYS)
+
+    # read shapefile into geopandas dataframe
+    shp_file = gpd.read_file(TEST_SHAPEFILE_PATH)
+
+    item_attrs = {}
+    for _index, row in shp_file.iterrows():
+        # convert each reach into a dictionary of attributes
+        item_attrs = json.loads(row.to_json(
+            default_handler=str), parse_float=Decimal)
+
+        # write to the table
+        hydrocron_test_table.add_data(**item_attrs)
+
+    items = hydrocron_test_table.run_query(partition_key='71224100223')
+
+    assert items[0]['wse'] == Decimal(str(286.2983))
+
+
+def test_delete_item(dynamo_instance):
+    '''
+    Test delete an item
+    '''
+    if dynamo_instance.table_exists(TEST_TABLE_NAME):
+        dynamo_instance.delete_table(TEST_TABLE_NAME)
+
+    hydrocron_test_table = dynamo_instance.create_table(
+        TEST_TABLE_NAME,
+        DYNAMO_KEYS)
+
+    # read shapefile into geopandas dataframe
+    shp_file = gpd.read_file(TEST_SHAPEFILE_PATH)
+
+    item_attrs = {}
+    for _index, row in shp_file.iterrows():
+        # convert each reach into a dictionary of attributes
+        item_attrs = json.loads(row.to_json(
+            default_handler=str), parse_float=Decimal)
+
+        # write to the table
+        hydrocron_test_table.add_data(**item_attrs)
+
+    hydrocron_test_table.delete_item(
+        partition_key='71224100203', sort_key=Decimal(-999999999999.000))
+    assert hydrocron_test_table.table.item_count == 686
+
+
+def test_delete_table(dynamo_instance):
+    '''
+    Test delete table
+    '''
+
+    if dynamo_instance.table_exists(TEST_TABLE_NAME):
+        dynamo_instance.delete_table(TEST_TABLE_NAME)
+    else:
+        dynamo_instance.create_table(
+            TEST_TABLE_NAME,
+            DYNAMO_KEYS)
+
+        dynamo_instance.delete_table()
+
+    assert not dynamo_instance.table_exists(TEST_TABLE_NAME)
+
+
+@pytest.fixture(scope='session', name="dynamo_instance")
+def dynamo_instance_fixture():
+    '''
+    Set up a boto3 resource connection to a local dynamodb instance. 
+    Assumes Local DynamoDB instance installed and running. 
```

Comment:
I asked Vanesa about this too; can we use https://pypi.org/project/pytest-dynamodb/ to move the dynamodb local instance start/stop into the test setup?

Reply:
done